### DISCUSSION TITLE
Different behavior on `...` and `pass`

### DISCUSSION THREAD

**Discussion by Kludex** [microsoft/pyright#5840] [Q&A] [answered]
Hey there 👋 

I noticed that `pyright` behaves differently on `...` and `pass`, and I was just curious about the reason. Is there a reason for it?

Thanks!

**Comment by erictraut** [comment, marked as answer]
Yes, this is intentional. See [this documentation](https://microsoft.github.io/pyright/#/mypy-comparison?id=ellipsis-in-function-body).

A `...` body is not typically used in real code — at least not outside of type stubs or protocol definitions, but it's often used in code samples where the implementation (the body of the function) is unimportant for purposes of illustration.

**Reply by Kludex** [reply]
Thanks 🙏
